src/main/java/ogc/rs/processes/tilesMetaDataOnboarding/TilesMetaDataOnboardingProcess.java

```diff
@@ -0,0 +1,465 @@
+                        requestInput.put("pureTile", true);
+                        promise.complete(requestInput);
+                    }
+                })
+                .onFailure(err -> {
+                    // Handle failure
+                    promise.fail(COLLECTION_EXISTENCE_CHECK_FAILURE_MESSAGE + " :" + err.getMessage());
+                });
+
+        return promise.future();
+    }
+
+    /**
+     * Onboards tile metadata into the database. Depending on whether the collection is a pure tile collection,
+     * it inserts data into different tables.
+     * <p>
+     * This method performs the necessary database operations to onboard tile metadata. It handles both pure tile collections
+     * and collections that combine feature and tile data.
+     * </p>
+     *
+     * @param requestInput JSON object containing tile metadata information and other details.
+     * @return A {@link Future<Void>} indicating the success or failure of the onboarding process.
+     */
+    private Future<Void> onboardTileMetadata(JsonObject requestInput) {
+        return pgPool.withTransaction(sqlClient -> {
+            Promise<Void> promise = Promise.promise();
+            LOGGER.debug("Starting the onboarding process for tile metadata.");
+
+            boolean pureTile = requestInput.getBoolean("pureTile");
+            String collectionId = requestInput.getString("resourceId");
+            String title = requestInput.getString("title");
+            String description = requestInput.getString("description");
+            String crs = requestInput.getString("crs");
+            JsonArray bbox = requestInput.getJsonArray("bbox");
+            JsonArray temporal = requestInput.getJsonArray("temporal");
+            String accessPolicy = requestInput.getString("accessPolicy");
+            String userId = requestInput.getString("userId");
+            String collectionType = requestInput.getString("collectionType");
+            String tmsId = requestInput.getString("tms_id");
+            JsonArray pointOfOrigin = requestInput.getJsonArray("pointOfOrigin");
+
+            // Convert JsonArray to Double[] for bbox and pointOfOrigin
+            Double[] bboxArray = (bbox != null) ? bbox.stream()
+                    .map(obj -> obj instanceof Number ? ((Number) obj).doubleValue() : 0.0)
+                    .toArray(Double[]::new) : new Double[0];
+
+            Double[] pointOfOriginArray = (pointOfOrigin != null) ? pointOfOrigin.stream()
+                    .map(obj -> obj instanceof Number ? ((Number) obj).doubleValue() : 0.0)
+                    .toArray(Double[]::new) : new Double[0];
+
+            // Convert JsonArray to String[] for temporal
+            String[] temporalArray = (temporal != null) ? temporal.stream()
+                    .map(Object::toString)
+                    .toArray(String[]::new) : new String[0];
+
+            Future<Void> collectionDetailsFuture = Future.succeededFuture();
```

Comment:
The Future<Void> collectionDetailsFuture = Future.succeededFuture(); is initialized with success to handle conditional execution cleanly. This allows the method to proceed with subsequent database operations regardless of whether the pureTile flag is true or false. If pureTile is true, the Future is reassigned to include the necessary database inserts. If pureTile is false, the chain continues without those inserts. I used this approach because it maintains a linear and clean flow in the asynchronous code.